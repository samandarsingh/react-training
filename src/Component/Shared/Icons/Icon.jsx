import React from "react";
import PropTypes from "prop-types";
import styles from "./Icon.module.scss";
/**
 * 
 * @param {string} fileName 
 * @param {string} altTitle
 * @return Element
 */
const Icon = props => {
    const {fileName,altTitle} = props;
    return (        
        <img className={styles.iconImage} 
            src={`assets/Icons/${fileName}`}
            alt={altTitle}>
        </img>       
    );
}
Icon.propTypes = {
    fileName: PropTypes.string.isRequired,
    altTitle: PropTypes.string.isRequired
}
export default Icon;
