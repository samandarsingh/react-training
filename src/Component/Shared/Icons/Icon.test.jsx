import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Icon from "./Icon";
describe('icon component testing', () => {
    const props ={
        fileName: "facebook.png",
        altTitle: "facebook"
    }
    const wrapper = mount(<Icon {...props} />);
    it(`Icon component should be run`,()=> {        
        expect(wrapper).toBeTruthy();
    });        
    it(`renders correctly`, () => {    
        const tree = renderer.create(<Icon {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});