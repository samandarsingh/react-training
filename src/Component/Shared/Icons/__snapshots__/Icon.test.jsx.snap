// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon component testing renders correctly 1`] = `
<img
  alt="facebook"
  className="iconImage"
  src="assets/Icons/facebook.png"
/>
`;
